libs/core/src/lib/radio/radio-button/radio-button.component.ts

```diff
@@ -0,0 +1,213 @@
+     * by default value is set to 'default'
+     */
+    @Input()
+    set state(newState: stateType) {
+        this._state = newState;
+        this.buildComponentCssClass();
+    }
+
+    private _disabled: boolean = false;
+    /** The field is used to tell if radio button should be disabled
+     * Value: true or false
+     * by default disabled state is set to false
+     */
+    @Input()
+    set disabled(isDisabled: boolean) {
+        this._disabled = isDisabled;
+        this.buildComponentCssClass();
+    }
+
+    /** @hidden */
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    /** The field should be only used with reactive forms
+     * Its purpose is to pass a current selected value from froumGroup
+     * The field is mandatory when working with reactive forms
+     */
+    @Input()
+    set selectedValue(v: any) {
+        this.actualValue = v;
+    }
+
+    private _name: string;
+    /** The name of the radio button
+     * The field is mandatory
+     */
+    @Input()
+    set name(v: string) {
+        this._name = v;
+    }
+
+    /** @hidden */
+    get name(): string {
+        return this._name;
+    }
+
+    get checked(): boolean {
+        if (this.value === undefined) {
+            return false;
+        }
+        return this.actualValue === this.value;
+    }
+
+    /**
+     * Set uniqueId to a radio button
+     */
+    @Input()
+    id: string = `radio-id-${uniqueId++}`;
+
+    /** Value field stores information about holding value by radio button
+     * The field is mandatory
+     */
+    @Input()
+    value: any;
+
+    // ControlValueAccessor implementation
+    /** @hidden */
+    onChange: any = () => {};
+
+    /** @hidden */
+    onTouched: any = () => {};
+
+    /** @hidden */
+    registerOnChange(fn: (selected: any) => { void }): void {
+        this.onChange = fn;
+    }
+
+    /** @hidden */
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+        this.changeDetectionRef.detectChanges();
+    }
+
+    /** @hidden */
+    writeValue(value: any): void {
+        this.valueChange(value);
+    }
+    // End implementation
+
+    /** @hidden */
+    labelClicked(): void {
+        this.valueChange(this.value);
+        this._setFocusOnNativeElement();
+    }
+
+    /** @hidden */
+    valueChange(value: any): void {
+        this.actualValue = value;
+
+        this._setNativeElementCheckedState();
+
+        this.changeDetectionRef.detectChanges();
+        this.onChange(value);
+    }
+
+    /** @hidden */
+    constructor(private changeDetectionRef: ChangeDetectorRef) {}
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this.buildComponentCssClass();
+        this._checkMandatoryFields();
+    }
+
+    /** @hidden */
+    @applyCssClass
+    buildComponentCssClass(): string {
```

Comment:
Could you add some comment, that this changes classes on the `fd-radio` input element?

Reply:
done